test/test_tensordict.py

```diff
@@ -1438,6 +1438,37 @@ def test_getitem_batch_size():
         assert expected_shape == resulting_shape, idx
 
 
+@pytest.mark.parametrize("device", get_available_devices())
+def test_requires_grad(device):
+    torch.manual_seed(1)
+    # Just one of the tensors have requires_grad
+    tensordicts = [
+        TensorDict(
+            batch_size=[11, 12],
+            source={
+                "key1": torch.randn(
+                    11, 12, 5, device=device, requires_grad=True if i == 5 else False
+                ),
+                "key2": torch.zeros(
+                    11, 12, 50, device=device, dtype=torch.bool
+                ).bernoulli_(),
+            },
+        )
+        for i in range(10)
+    ]
+    sub_td = LazyStackedTensorDict(*tensordicts, stack_dim=0)
```

Comment:
maybe `stacked_td` instead of `sub_td`?
You could try to index the tensordict too by the way:
```python
sub_td1 = td[0]
assert sub_td1.get("key1").requires_grad
assert sub_td1.get_meta("key1").requires_grad

sub_td2 = td[torch.tensor([1, 2])]
assert sub_td2.get("key1").requires_grad
assert sub_td2.get_meta("key1").requires_grad
```

We could also do the same for unsqueeze and/or view?

Reply:
I've added test class with different TensorDicts with requires_grad and some operations like view and squeeze.